docs/Tips-N'-Tricks.md

```diff
@@ -140,3 +140,42 @@ export HOMEBREW_ARTIFACT_DOMAIN=https://artifacts.example.com/artifactory/homebr
 export HOMEBREW_ARTIFACT_DOMAIN_NO_FALLBACK=1
 export HOMEBREW_DOCKER_REGISTRY_BASIC_AUTH_TOKEN="$(printf 'anonymous:' | base64)"
 ```
+
+## Loading Homebrew from the same dotfiles on different operating systems
+
+Some users may want to use the same shell initialization files on macOS and Linux.
+Use this to detect the likely Homebrew installation directory and load Homebrew when it's found.
+You may need to adapt this to your particular shell or other particulars of your environment.
+
+```sh
+# Execute only if brew isn't already available.
+if ! [ -x "$(command -v brew)" ]; then
+  OS="$(uname)"
+  UNAME_MACHINE="$(uname -m)"
+  if [ "${OS}" = "Linux" ]; then
+    # Linux
+    HOMEBREW_PREFIX="/home/linuxbrew/.linuxbrew"
+  elif [ "${OS}" = "Darwin" ]; then
+    if [ "${UNAME_MACHINE}" = "arm64" ]; then
+      # M-series ARM64 macOS
+      HOMEBREW_PREFIX="/opt/homebrew"
+    else
+      # Intel macOS
+      HOMEBREW_PREFIX="/usr/local"
+    fi
+  fi
```

Comment:
I do this in my cross-platform startup files:
```
  case "$(uname -sm)" in
    # Intel macOS
    "Darwin x86_64") HOMEBREW_PREFIX="/usr/local";;
    # M-series ARM64 macOS
    "Darwin arm64")  HOMEBREW_PREFIX="/opt/homebrew";;
    # Linux
    "Linux "*)       HOMEBREW_PREFIX="/home/linuxbrew/.linuxbrew";;
    *) echo "Warning: Homebrew is not supported on this platform." >&2;;
  esac
```
The default case is optional, of course.

Reply:
Ooh, yeah, I should golf my example down to that.

When I set this up for myself several few years ago, I checked directories' and files' existence instead of looking at uname output: https://github.com/colindean/hejmo/blame/35b40129563c27be051d316cc67a7fdcccb93daf/bash_profile/00_path.sh#L4-L18

(I'm also looking at that file casually for the first time in many years and realizing there's more stuff hardcoded to /usr/local than should be!)

Reply:
Actually, it may not: fish's `case` syntax is different than zsh/bash, and the above is POSIX-compliant, good enough for docs.